Framework/Reflectometry/src/ReflectometryReductionOneAuto3.cpp

```diff
@@ -842,144 +853,163 @@ bool ReflectometryReductionOneAuto3::processGroups() {
+    groupAlg->setProperty("OutputWorkspace", outputNames.iVsQBinned);
+    groupAlg->execute();
+  }
+  if (anyWorkspaceInListExists(IvsLamGroup)) {
     groupAlg->setProperty("InputWorkspaces", IvsLamGroup);
-    groupAlg->setProperty("OutputWorkspace", output.iVsLam);
+    groupAlg->setProperty("OutputWorkspace", outputNames.iVsLam);
     groupAlg->execute();
   }
-  groupAlg->setProperty("InputWorkspaces", IvsQBinnedGroup);
-  groupAlg->setProperty("OutputWorkspace", output.iVsQBinned);
-  groupAlg->execute();
-  groupAlg->setProperty("InputWorkspaces", IvsQGroup);
-  groupAlg->setProperty("OutputWorkspace", output.iVsQ);
-  groupAlg->execute();
 
-  // Set other properties so they can be updated in the Reflectometry interface
-  setPropertyValue("ThetaIn", alg->getPropertyValue("ThetaIn"));
-  setPropertyValue("MomentumTransferMin",
-                   alg->getPropertyValue("MomentumTransferMin"));
-  setPropertyValue("MomentumTransferMax",
-                   alg->getPropertyValue("MomentumTransferMax"));
-  setPropertyValue("MomentumTransferStep",
-                   alg->getPropertyValue("MomentumTransferStep"));
-  setPropertyValue("ScaleFactor", alg->getPropertyValue("ScaleFactor"));
+  setPropertyValue("OutputWorkspace", outputNames.iVsQ);
+  setPropertyValue("OutputWorkspaceBinned", outputNames.iVsQBinned);
+  setPropertyValue("OutputWorkspaceWavelength", outputNames.iVsLam);
+}
+
+void ReflectometryReductionOneAuto3::setOutputProperties(
+    OutputProperties const &outputProperties) {
+  setProperty("ThetaIn", outputProperties.thetaIn);
+  setProperty("MomentumTransferMin", outputProperties.qMin);
+  setProperty("MomentumTransferMax", outputProperties.qMax);
+  setProperty("MomentumTransferStep", outputProperties.qStep);
+  setProperty("ScaleFactor", outputProperties.scale);
+}
 
-  setOutputWorkspaces(output, IvsLamGroup, IvsQBinnedGroup, IvsQGroup);
+/** Process groups. Groups are processed differently depending on transmission
+ * runs and polarization analysis.
+ */
+bool ReflectometryReductionOneAuto3::processGroups() {
+  // this algorithm effectively behaves as MultiPeriodGroupAlgorithm
+  m_usingBaseProcessGroups = true;
 
+  // Get our input workspace group
+  auto group = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(
+      getPropertyValue("InputWorkspace"));
+  // Get the run number for use in constructing default names
+  std::string runNumber = getRunNumberForWorkspaceGroup(group);
+
+  // Execute child algorithm over each group member and collate the output
+  // workspace names and output properties
+  std::vector<std::string> IvsQBinnedGroup, IvsQGroup, IvsLamGroup;
+  OutputProperties outputProperties;
+  for (size_t i = 0; i < group->size(); ++i) {
+    auto inputName = group->getItem(i)->getName();
+    auto outputNames = getOutputNamesForGroupMember(inputName, runNumber, i);
+    auto alg = createAlgorithmForWorkspaceInGroup(inputName, outputNames);
+    alg->execute();
```

Comment:
By default algorithms do not rethrow an exception and instead catch it on the callers behalf silently. It might be worth adding in `createAlgorithmForWorkspaceInGroup`:

`alg->setRethrows(true);`

So that if something does go awry this parent alg will stop and unwind with the exception

Reply:
That's already being set in that function (it doesn't show in the diffs because it was always there)